File: src/main/java/sprykerFileCreator/model/definition/spryker/ZedPluginDefinition.java
package pav.sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ZedPluginDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.ZED_PLUGIN;
    }

    @Override
    public String getNamePattern() {
        return "{{ className }}Plugin";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Zed\\{{ bundleName }}\\Communication\\Plugin";
    }

    @Override
    public String getMethodForReturnType() {
        return "";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.ZED_COMMUNICATION_FACTORY,
                SprykerConstants.ZED_FACADE,
                SprykerConstants.ZED_CONFIG,
                SprykerConstants.ZED_QUERY_CONTAINER,
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Zed\\Kernel\\Communication\\AbstractPlugin";
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/ZedPluginDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class ZedPluginDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.ZED_PLUGIN
    override val namePattern: String
        get() = "{{ className }}Plugin"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Zed\\{{ bundleName }}\\Communication\\Plugin"
    override val methodForReturnType: String
        get() = ""
    override val docBlockClasses: Array<String>
        get() = arrayOf(
                SprykerConstants.ZED_COMMUNICATION_FACTORY,
                SprykerConstants.ZED_FACADE)
    override val defaultParentClass: String
        get() = "Spryker\\Zed\\Kernel\\Communication\\AbstractPlugin"
}
